server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java

```diff
@@ -19,179 +19,211 @@
-      .addAggregation(AggregationBuilders.terms(ActiveRuleNormalizer.ActiveRuleField.PROFILE_KEY.field())
-        .field(ActiveRuleNormalizer.ActiveRuleField.PROFILE_KEY.field()).size(0)
-        .subAggregation(AggregationBuilders.terms(ActiveRuleNormalizer.ActiveRuleField.INHERITANCE.field())
-          .field(ActiveRuleNormalizer.ActiveRuleField.INHERITANCE.field()))
-        .subAggregation(AggregationBuilders.terms(ActiveRuleNormalizer.ActiveRuleField.SEVERITY.field())
-          .field(ActiveRuleNormalizer.ActiveRuleField.SEVERITY.field()))
+          .mustNot(FilterBuilders.hasParentFilter(TYPE_RULE,
+            FilterBuilders.termFilter(FIELD_RULE_STATUS, RuleStatus.REMOVED.name())))))
+      .addAggregation(AggregationBuilders.terms(FIELD_ACTIVE_RULE_PROFILE_KEY)
+        .field(RuleIndexDefinition.FIELD_ACTIVE_RULE_PROFILE_KEY).size(0)
+        .subAggregation(AggregationBuilders.terms(FIELD_ACTIVE_RULE_INHERITANCE)
+          .field(RuleIndexDefinition.FIELD_ACTIVE_RULE_INHERITANCE))
+        .subAggregation(AggregationBuilders.terms(FIELD_ACTIVE_RULE_SEVERITY)
+          .field(RuleIndexDefinition.FIELD_ACTIVE_RULE_SEVERITY))
         .subAggregation(AggregationBuilders.count(COUNT_ACTIVE_RULES)))
       .setSize(0)
-      .setTypes(this.getIndexType());
+      .setTypes(TYPE_ACTIVE_RULE);
     SearchResponse response = request.get();
     Map<String, Multimap<String, FacetValue>> stats = new HashMap<>();
-    Aggregation aggregation = response.getAggregations().get(ActiveRuleNormalizer.ActiveRuleField.PROFILE_KEY.field());
+    Aggregation aggregation = response.getAggregations().get(FIELD_ACTIVE_RULE_PROFILE_KEY);
     for (Terms.Bucket value : ((Terms) aggregation).getBuckets()) {
-      stats.put(value.getKey()
-        , this.processAggregations(value.getAggregations()));
+      stats.put(value.getKey(), processAggregations(value.getAggregations()));
     }
 
     return stats;
   }
+
+  private Multimap<String, FacetValue> processAggregations(Aggregations aggregations) {
+    Multimap<String, FacetValue> stats = ArrayListMultimap.create();
+    if (aggregations != null) {
+      for (Aggregation aggregation : aggregations.asList()) {
+        if (aggregation instanceof StringTerms) {
+          for (Terms.Bucket value : ((Terms) aggregation).getBuckets()) {
+            FacetValue facetValue = new FacetValue(value.getKey(), value.getDocCount());
+            stats.put(aggregation.getName(), facetValue);
+          }
+        } else if (aggregation instanceof InternalValueCount) {
+          InternalValueCount count = (InternalValueCount) aggregation;
+          FacetValue facetValue = new FacetValue(count.getName(), count.getValue());
+          stats.put(count.getName(), facetValue);
+        }
+      }
+    }
+    return stats;
+  }
+
+  private enum ToDoc implements Function<Map<String, Object>, ActiveRuleDoc> {
+    INSTANCE;
+
+    @Override
+    public ActiveRuleDoc apply(@Nonnull Map<String, Object> input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)